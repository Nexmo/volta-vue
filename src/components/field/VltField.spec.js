import { mount } from '@vue/test-utils';
import { VltField } from '..';

describe('vlt-field', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(VltField);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('has label "Test field"', async () => {
    wrapper.setProps({ label: 'Test field' });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is optional', async () => {
    wrapper.setProps({ optional: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is big', async () => {
    wrapper.setProps({ big: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is big with label "Test field"', async () => {
    wrapper.setProps({
      big: true,
      label: 'Test field',
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('has tooltip "Hello World"', async () => {
    wrapper.setProps({
      label: 'Test field',
      tooltip: 'Hello World',
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('has error "something is wrong"', async () => {
    wrapper.setProps({ error: 'something is wrong' });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });
});
