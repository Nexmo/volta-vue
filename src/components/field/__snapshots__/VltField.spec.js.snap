// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-field has error "something is wrong" 1`] = `
<div
  class="Vlt-form__element Vlt-form__element--error"
  id=""
>
  <!---->
    
  <small
    class="Vlt-form__element__error"
    id=""
  >
    something is wrong
  </small>
</div>
`;

exports[`vlt-field has label "Test field" 1`] = `
<div
  class="Vlt-form__element"
  id=""
>
  <label
    class="Vlt-label"
  >
    Test field
    
    <!---->
     
    <!---->
  </label>
    
  <!---->
</div>
`;

exports[`vlt-field has tooltip "Hello World" 1`] = `
<div
  class="Vlt-form__element"
  id=""
>
  <label
    class="Vlt-label"
  >
    Test field
    
    <!---->
     
    <vlt-tooltip-stub
      title="Hello World"
    />
  </label>
    
  <!---->
</div>
`;

exports[`vlt-field is big 1`] = `
<div
  class="Vlt-form__element Vlt-form__element--big"
  id=""
>
  <!---->
    
  <!---->
</div>
`;

exports[`vlt-field is big with label "Test field" 1`] = `
<div
  class="Vlt-form__element Vlt-form__element--big"
  id=""
>
  <!---->
    
  <!---->
</div>
`;

exports[`vlt-field is optional 1`] = `
<div
  class="Vlt-form__element"
  id=""
>
  <!---->
    
  <!---->
</div>
`;

exports[`vlt-field renders correctly 1`] = `
<div
  class="Vlt-form__element"
  id=""
>
  <!---->
    
  <!---->
</div>
`;
